app/renderer/lib/appium-client.js

```diff
@@ -221,90 +227,79 @@ export default class AppiumClient {
-        pixelRatio,
-        platformName,
-        statBarHeight,
-        viewportRect,
-        webViewPosition;
+  // Retrieve all detected contexts, as well as the current context
+  // If retrieval of either one fails, return the error(s)
+  // Additionally, if webview is used, adjust the found element positions to fit screenshot
+  // Only called while in hybrid mode
+  async getContextUpdate ({ windowSize }) {
+    let contexts, contextsError, currentContext, currentContextError, webviewTopOffset;
+    let webviewLeftOffset = 0;
+
     if (!await this.hasContextsCommand()) {
       return {currentContext: null, contexts: []};
     }
 
+    // First get the current context (or the error, if one appears)
     try {
       currentContext = await this.driver.getContext();
     } catch (e) {
       currentContextError = e;
     }
 
-    // Note: These methods need to be executed in the native context because ChromeDriver behaves differently
+    // The retrieval of all contexts and webview position adjustments require some native context use
     if (currentContext !== NATIVE_APP) {
       await this.driver.switchContext(NATIVE_APP);
     }
 
-    ({platformName, pixelRatio, statBarHeight, viewportRect} = await this.driver.getSession());
+    const { platformName, statBarHeight, viewportRect } = await this.driver.getSession();
     const isAndroid = _.toLower(platformName) === 'android';
 
+    // Get all available contexts (or the error, if one appears)
     try {
       contexts = await this.driver.executeScript('mobile:getContexts', []);
       contexts = isAndroid ? this.parseAndroidContexts(contexts) : contexts;
     } catch (e) {
       contextsError = e;
     }
 
-
+    // For webview context, the viewport needs to be recalculated
+    // to account for any top and left offsets
     if (currentContext !== NATIVE_APP) {
-      try {
-        // Get the webview offset
-        if (viewportRect) {
-          // The viewport rectangles are based on the screen density,
-          // iOS needs CSS pixels
-          webViewPosition = {
-            x: isAndroid ? viewportRect.left : Math.round(viewportRect.left / pixelRatio),
-            y: isAndroid ? viewportRect.top : Math.round(viewportRect.top / pixelRatio),
-          };
-        } else {
-          // Fallback
-          const el = await this.driver.findElement(
-            isAndroid ? 'xpath' : '-ios class chain',
-            isAndroid ? '//android.webkit.WebView' : '**/XCUIElementTypeWebView'
-          );
-          if (el) {
-            webViewPosition = await el.getRect();
-          }
+      if (isAndroid) {
+        // on Android, find the root webview element and use its X and Y startpoints
+        const webview = await this.fetchElement({strategy: 'xpath', selector: ANDROID_WEBVIEW_SELECTOR});
+        if (webview.el) {
+          const { x, y } = await webview.el.getRect();
+          webviewTopOffset = y;
+          webviewLeftOffset = x;
+        }
+      } else {
```

Comment:
It's probably not a big deal right now cause I'm sure there are lots of these sprinkled in the codebase, but moving forward we should try to take care that we don't assume sessions will always be android or ios. This `else` clause could run in a Roku TV session, for example, where it would be irrelevant.